patients/update.go

```diff
@@ -72,53 +72,154 @@ func CreateSummaryUpdateBody(summary *summaries.Summary) clinics.UpdatePatientSu
 		LastData:                 summary.LastData,
 		LastUpdatedDate:          summary.LastUpdatedDate,
 		LastUploadDate:           summary.LastUploadDate,
+		HasLastUploadDate:        summary.HasLastUploadDate,
 		OutdatedSince:            summary.OutdatedSince,
-		TotalDays:                summary.TotalDays,
+		TotalHours:               summary.TotalHours,
 		LowGlucoseThreshold:      summary.LowGlucoseThreshold,
 		VeryLowGlucoseThreshold:  summary.VeryLowGlucoseThreshold,
 		HighGlucoseThreshold:     summary.HighGlucoseThreshold,
 		VeryHighGlucoseThreshold: summary.VeryHighGlucoseThreshold,
 	}
 
-	var periodExists = false
-	var period14dExists = false
 	if summary.Periods != nil {
-		periodExists = true
+		patientUpdate.Periods = &clinics.PatientSummaryPeriods{}
+
+		if summary.Periods.N1d != nil {
+			patientUpdate.Periods.N1d = &clinics.PatientSummaryPeriod{
+				AverageGlucose: &clinics.AverageGlucose{Value: summary.Periods.N1d.AvgGlucose.Value,
+					Units: clinics.AverageGlucoseUnits(summary.Periods.N1d.AvgGlucose.Units)},
+				GlucoseManagementIndicator:    summary.Periods.N1d.GlucoseManagementIndicator,
+				HasGlucoseManagementIndicator: summary.Periods.N1d.HasGlucoseManagementIndicator,
+
+				TimeCGMUseMinutes:    summary.Periods.N1d.TimeCGMUseMinutes,
```

Comment:
In retrospect I wish we had defined a common shared data model for these fields, something like this (in OAS3 YAML):

```
title: TargetTime
description: >-
  Describes time on target.
type: object
properties:
  minutes:
    type: int
  records:
    type: int
  percent:
    type: number
required:
  - minutes
  - records
```

Then we could have used that for every one of these top-level fields and make it easier to copy the fields from one object to another.

Reply:
Sadly, im not sure how this would be translated to golang due to how the typing and the codegen works, closing for now.